internal/locate/region_cache.go

```diff
@@ -395,7 +398,13 @@ type RegionCache struct {
 	testingKnobs struct {
 		// Replace the requestLiveness function for test purpose. Note that in unit tests, if this is not set,
 		// requestLiveness always returns unreachable.
-		mockRequestLiveness func(s *Store, bo *retry.Backoffer) livenessState
+		mockRequestLiveness atomic.Value
+	}
+
+	regionsNeedReload struct {
```

Comment:
How about using a channel so `Mutex` could be saved and operations on the `RegionCache` are already synchronized?